### ISSUE TITLE
SDK crash on Samsung devices (Android 7)

### ISSUE THREAD

**Issue by cpetzel** [onfido/onfido-android-sdk#32]
#### What was the expected behaviour?
SDK does not crash when taking a picture on Samsung devices

#### What happened instead?
SDK crashes. 
See this link for more info
http://crashes.to/s/763412d2eae

```
java.lang.NullPointerException Attempt to invoke virtual method 'void com.onfido.android.sdk.capture.ui.camera.face.CameraSource.autoFocusOnce(java.util.List, com.onfido.android.sdk.capture.ui.camera.face.CameraSource$AutoFocusCallback)' on a null object reference 
    Unknown:-1 com.onfido.android.sdk.capture.ui.camera.face.CameraSourcePreview$1.onClick
    View.java:6312 android.view.View.performClick
    View.java:23985 android.view.View$PerformClick.run
    Handler.java:751 android.os.Handler.handleCallback
    Handler.java:95 android.os.Handler.dispatchMessage
    Looper.java:154 android.os.Looper.loop
    ActivityThread.java:6816 android.app.ActivityThread.main
    Method.java:-2 java.lang.reflect.Method.invoke
    ZygoteInit.java:1563 com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run
    ZygoteInit.java:1451 com.android.internal.os.ZygoteInit.main
```

#### Version info:

- `onfido-capture-sdk` version:
compile "com.onfido.sdk.capture:onfido-capture-sdk:4.3.0"

- Android API version(s): 7 & 7.1

- `compileSdkVersion`: 27
- `targetSdkVersion`: 27
- `minSdkVersion`: 18
- Support library version (if applicable):

#### Steps to reproduce:
Samsung - SM-J510FNSM-J5 on Android 7.1.1
Samsung - SM-G610F on Android 7

#### Other useful info:
_screenshots, videos, etc._

**Comment by sergiomarquesmoura** [comment]
Hi @cpetzel . Thanks for flagging this, we will try to replicate it with a similar device.

**Comment by sergiomarquesmoura** [comment]
Hi @cpetzel . We're trying to replicate this on a Samsung Galaxy S9 running Android 8.0. However, we couldn't yet. In case you have been able to reproduce it, could you provide us the steps?

**Comment by cpetzel** [comment]
Information on this crash was gathered from our crash reporting tool, with real users using our production app. Unfortunately, I don't have any more information about how to reproduce. 

- Perhaps you can try with `don't keep activities` developer option? 

- Try overloading the resources on a device when using an app with your SDK ?
https://github.com/stressapptest/stressapptest

**Comment by cpetzel** [comment]
As the title of the bug depicts, this crash in happening on Android 7 and 7.1 for this specific device. A lot of our users are not on the latest and greatest hardware (S9). We would expect your SDK to work on all versions of android devices. 

I would suggest that you get your hands on a specific Samsung device, such as [this](https://www.amazon.com/Samsung-Galaxy-Prime-G610F-DS/dp/B01N4WGN48) and fix the errors. 

Thanks!

**Comment by sergiomarquesmoura** [comment]
Hi @cpetzel . Thanks for flagging this! 
We just released `4.5.1` which addresses this issue. Please try it out when you have a chance.

**Comment by cpetzel** [comment]
Thanks @sergiomarquesmoura . I will update and see if it resolves our crash rate over the next few weeks.
